packages/core/src/discovery.js

```diff
@@ -69,6 +69,11 @@ function debugSnapshotOptions(snapshot) {
   debugProp(snapshot, 'clientInfo');
   debugProp(snapshot, 'environmentInfo');
   debugProp(snapshot, 'domSnapshot', Boolean);
+  if (snapshot.responsiveSnapshotCapture) {
+    debugProp(snapshot, 'domSnapshot.0.userAgent');
```

Comment:
this can fail if user has set `responsiveSnapshotCapture` in cli yml but is using older sdk which wont use this. 
Update condition to check if domSnapshot is a list or not

Reply:
Also I am assuming that you are not using this value yet and it will be a different pr. but we need to use this in default requestHeaders

Reply:
yes that will be in separate PR as this PR already has lots of things